### ISSUE TITLE
stb_image and stb_image_writer can't be found by compiler

### ISSUE THREAD

**Issue by mioxyz** [nsmryan/wfc-rs#1]
Hi, I'm trying to build your RustRoguelike project, however I'm having issues with wfc. 
in an otherwise empty project, I included wfc-rs, like so:
```
[package]
name = "wfc_test"
version = "0.1.0"
[dependencies]
wfc-rs = "0.5"
```
which produces the following output:
```
cargo run                                                                                                                               
    Updating crates.io index
  Downloaded libc v0.2.121
  Downloaded wfc-rs v0.5.1
  Downloaded cc v1.0.73
  Downloaded 3 crates (737.8 KB) in 0.39s
   Compiling cc v1.0.73
   Compiling libc v0.2.121
   Compiling wfc-rs v0.5.1
The following warnings were emitted during compilation:

warning: In file included from csrc/wfc.c:7:
warning: wfc/wfc.h:159:10: fatal error: stb_image.h: No such file or directory
warning:   159 | #include "stb_image.h"
warning:       |          ^~~~~~~~~~~~~
warning: compilation terminated.

error: failed to run custom build command for `wfc-rs v0.5.1`

Caused by:
  process didn't exit successfully: `/mnt/home/mio/code/rust/RustTest/target/debug/build/wfc-rs-fb03647340e75b31/build-script-build` (exit status: 1)
  --- stdout
  TARGET = Some("x86_64-unknown-linux-gnu")
  OPT_LEVEL = Some("0")
  HOST = Some("x86_64-unknown-linux-gnu")
  CC_x86_64-unknown-linux-gnu = None
  CC_x86_64_unknown_linux_gnu = None
  HOST_CC = None
  CC = None
  CFLAGS_x86_64-unknown-linux-gnu = None
  CFLAGS_x86_64_unknown_linux_gnu = None
  HOST_CFLAGS = None
  CFLAGS = None
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("true")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  running: "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-I" "wfc/" "-Wall" "-Wextra" "-o" "/mnt/home/mio/code/rust/RustTest/target/debug/build/wfc-rs-d3fc4ad0d8afc8b4/out/csrc/wfc.o" "-c" "csrc/wfc.c"
  cargo:warning=In file included from csrc/wfc.c:7:
  cargo:warning=wfc/wfc.h:159:10: fatal error: stb_image.h: No such file or directory
  cargo:warning=  159 | #include "stb_image.h"
  cargo:warning=      |          ^~~~~~~~~~~~~
  cargo:warning=compilation terminated.
  exit status: 1
  --- stderr
  error occurred: Command "cc" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-I" "wfc/" "-Wall" "-Wextra" "-o" "/mnt/home/mio/code/rust/RustTest/target/debug/build/wfc-rs-d3fc4ad0d8afc8b4/out/csrc/wfc.o" "-c" "csrc/wfc.c" with args "cc" did not execute successfully (status code exit status: 1).
```
For some reason the c compiler is having problems finding the `stb_image.h` and `stb_image_writer.h` header files. I tried building [krychu/wfc](https://github.com/krychu/wfc) by placing the two files into the source directory and that builds just fine.

I'm running Manjaro Linux on a thinkpad and I can't program in Rust.

**Comment by nsmryan** [comment]
That is strange. It happens that I also have a Manjaro Linux installation, so I tried on my laptop, and I got the same error.

I fixed this problem, and some other ones that appear to occur only on Linux (gcc reporting some warnings as errors that I just suppressed). The result is published to crates.io as version 5.2. I'm not really sure why cargo reports the error as it does, since I fixed a different error, but things seem to work now.

I pushed a version of RustRoguelike with the 5.2 dependency and it builds for Manjaro Linux for me.

By the way, thank you for your interest! The game is very much in development, so its not really a playable game, but it might be fun to at least open up and try a little bit.
